### PR TITLE
Upgrade the static site generation engine, hash CSS and JS and remove IE8 support

### PR DESCRIPTION
**Pull request by kr8n3r**
## What

Upgrade the static site generation engine, hash CSS and JS and remove IE8 support 

- [x] stop serving IE8 specific stylesheet
- [x] move from `next-mdx-enhanced` to `next-mdx-remote`
- [x] add hashes to static assets (css&js)
- [x] prevent NextJS scripts from loading

## Why 

`next-mdx-enhanced` is being left behind and move to `next-mdx-remote` is encouraged as the former is not compatible with newer version of Nextjs. We want to be able to upgrade. We're taking this opportunity to simplify the templates and drop IE8 support.

## How to review

- **suggest reviewing by commit**
- view deployed work on https://paas-product-pages-rework.london.cloudapps.digital/
**or**
- clone branch
- run `npm install`
- run the site locally: `npm run local` to check pages load and assets have hashes
- build the site `npm run build` to check production build is fine 
- `cd` into `out` folder and serve the page (`npx serve`)
- check page on `localhost:5000` for any issues

Screenshot 2021-08-02 at 09 24 33

### CODE REVIEW

File: pages/[slug].tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+export const getStaticPaths:GetStaticPaths = async () => {
+  const paths = fs
+    .readdirSync(contentFilePath)
+    .filter((path) => /\.mdx?$/.test(path))
+    // Remove file extensions for page paths
+    .map((path) => path.replace(/\.mdx?$/, ''))
+    // exclude homepage as it has its own template
+    .filter((path) => !path.includes('index'))
+    // Map the path into the static paths object required by Next.js
+    .map((slug) => ({ params: { slug } }))
```
Comment (0atman) [inline pages/[slug].tsx:122]: Big fan of this chain pattern - however I think the need to comment them might suggest that these functions could do with names?
How about something like:

```js
const remove_mdx = path => path.replace(/\.mdx?$/, '')
const paths_without_index = path => !path.includes('index')
const paths_object = slug => ({ params: { slug } })
```

File: pages/_document.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import Document, { Head, Html, Main, NextScript } from 'next/document'
+import CookieBanner from '@components/CookieBanner'
+import Footer from '@components/Footer'
+import Header from '@components/Header'
+import javascriptHash from '../public/assets/javascript/manifest.json'
+
+class GovukTemplate extends Document {
+  static async getInitialProps (ctx) {
+    const initialProps = await Document.getInitialProps(ctx)
```
Comment (0atman) [inline pages/_document.tsx:9]: Couldn't this be called `context`?
I'm sure this is probably some standard I'm not aware of yet, so please ignore me!

File: pages/content/cookies.mdx [hunk 3]
```diff
@@ -2,18 +2,18 @@
...
-  <div class="govuk-notification-banner__content">
-    <p class="govuk-notification-banner__heading">
+  <div className="govuk-notification-banner__content">
+    <p className="govuk-notification-banner__heading">
       Your cookie settings were saved
     </p>
-    <p class="govuk-body">
-      <a class="govuk_link previous-page govuk-!-margin-top-1" href="#">
+    <p className="govuk-body">
+      <a className="govuk_link previous-page govuk-!-margin-top-1" href="#">
```
Comment (0atman) [inline pages/content/cookies.mdx:16]: I'd be interested to know what the `className` pattern here is for?

**Review by 0atman — CHANGES_REQUESTED** [review]
Sorry this took a while, the only change I think could be nice is the chain pattern naming of the anonymous functions :ok_hand:

File: pages/content/cookies.mdx (hunk 3 above)
Comment (kr8n3r) [reply pages/content/cookies.mdx:16]: it's still React so complier will throw an warning. :)

File: pages/_document.tsx (hunk 2 above)
Comment (kr8n3r) [reply pages/_document.tsx:9]: going by their Custom Documents docs https://github.com/vercel/next.js/blob/canary/docs/advanced-features/custom-document.md#custom-document

File: pages/[slug].tsx (hunk 1 above)
Comment (kr8n3r) [reply pages/[slug].tsx:122]: hmm these are not anonymous functions. 
it's map and filter methods on the Array class and `path => !path.includes('index')` for example is a test 

not sure it makes sense to create another set of named functions to return matching results. do you have any good examples of this?

File: pages/_document.tsx (hunk 2 above)
Comment (0atman) [reply pages/_document.tsx:9]: got it, many thanks!

File: pages/content/cookies.mdx (hunk 3 above)
Comment (0atman) [reply pages/content/cookies.mdx:16]: aha! cool

File: pages/[slug].tsx (hunk 1 above)
Comment (0atman) [reply pages/[slug].tsx:122]: They are equivalent, right? https://www.w3schools.com/Js/js_arrow_function.asp

Comment (0atman) [reply pages/[slug].tsx:122]: Cool!

**Review by 0atman — APPROVED** [review]
thanks for the changes! All good!

**Pull request merged**